File: Week-02/day-03/diagonal-matrix.py
# - Create (dynamically) a two dimensional list
#   with the following matrix. Use a loop!
#
#   1 0 0 0
#   0 1 0 0
#   0 0 1 0
#   0 0 0 1
#
# - Print this two dimensional list to the output

a = int(input('Elements of the matrix: '))

print("\n")
Matrix = [[0 for x in range(a)] for y in range(a)]
for i in range(0, a):
    Matrix[i][i] = 1

for i in range(0, a):
    print(Matrix[i][:])

print("\n Same, but joined: \n")

for row in Matrix:
  print(' '.join(map(str,row)))

File: Week-07/Day-03/diagonal-matrix.js
'use strict';
// - Create (dynamically*) a two dimensional list
//   with the following matrix**. Use a loop!
//
//   0 0 0 1
//   0 0 1 0
//   0 1 0 0
//   1 0 0 0
//
// - Print this two dimensional list to the console
//
// * size should depend on a variable
// ** Relax, a matrix is just like an array

var rows = 10;

for (var i = 1; i <= rows; i++) {
  var line = '';
  for (var j = 0; j < rows; j++) {
    if (j === rows-i) {
      line += ' 1'
    } else {
      line += ' 0'
    }
  }
  console.log( line );
}
